macaroons/service.go

```diff
@@ -257,3 +261,46 @@ func (svc *Service) ListMacaroonIDs(_ context.Context) ([][]byte, error) {
 
 	return rootKeySlice, nil
 }
+
+// DeleteMacaroonID removes one specific root key ID. If the root key ID is
+// found and deleted, it will be returned.
+func (svc *Service) DeleteMacaroonID(
+	_ context.Context, rootKeyID []byte) ([]byte, error) {
+
+	svc.rks.encKeyMtx.RLock()
+	defer svc.rks.encKeyMtx.RUnlock()
+
+	// Check it's unlocked.
+	if svc.rks.encKey == nil {
+		return nil, ErrStoreLocked
+	}
+
+	// Deleting encryptedKeyID or DefaultRootKeyID is not allowed.
+	if bytes.Equal(rootKeyID, encryptedKeyID) ||
+		bytes.Equal(rootKeyID, DefaultRootKeyID) {
+		return nil, ErrDeletionForbidden
```

Comment:
nit: insert empty line after condition that spans multiple lines (for readability).

Reply:
my bad, fixed